crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -991,6 +992,14 @@ fn new_bridge(
 }
 
 pub fn chown_certificate_and_key(bridge_config: &BridgeConfig) {
+    // Skip chown when using Basic Auth
+    if bridge_config
+        .auth_method
+        .is_some_and(|auth_method| auth_method == AuthMethod::Basic)
```

Comment:
Actually, this condition doesn't cover when `auth_method == AuthMethod::Auto` and auto chooses basic. You could use `auth_method.is_basic(<path_to_credentials_toml>)` instead, however, you need to parse the credentials file path, which is not available in this function. 

Alternatively, if both `remote_username` and `remote_password` are set, you can consider that auth method is basic. 

```suggestion
    if bridge_config.remote_username.is_some() && bridge_config.remote_password.is_some() 
```

Reply:
When [Bridge Config](https://github.com/thin-edge/thin-edge.io/blob/8c2f74f9f9759982c260b4870ce0a734a0b34233/crates/core/tedge/src/bridge/c8y.rs#L151) is created, based on `remote_username` it is either `AuthMethod::Certificate` or `AuthMethod::Basic`. For `aws` and `azure` it is set to `None`.

Reply:
> When [Bridge Config](https://github.com/thin-edge/thin-edge.io/blob/8c2f74f9f9759982c260b4870ce0a734a0b34233/crates/core/tedge/src/bridge/c8y.rs#L151) is created, based on remote_username it is either AuthMethod::Certificate or AuthMethod::Basic. For aws and azure it is set to None.

This is correct. So, just to fix the issue, your change is enough.

However, there is a mess in the code. `BridgeConfig::auth_method` should be `Option<AuthType>`, `AuthType` has only `Basic` or `Certificate`. Furthermore, what is `None` mean? AWS/Azure, we use certificate based authentication. So, for me, even `BridgeConfig::auth_method` should be no longer Option, just `AuthType`.

Reply:
As the condition seems no so obvious, it would be better to extract it into a `BridgeConfig` method.